net.sourceforge.projects.vht.yml

```diff
@@ -0,0 +1,19 @@
+id: net.sourceforge.projects.vht
+runtime: org.gnome.Platform
+runtime-version: '46'
+sdk: org.gnome.Sdk
+command: vht
+finish-args:
+  - --filesystem=home
+  - --filesystem=xdg-run/pipewire-0
+  - --socket=wayland
+  - --socket=fallback-x11
```

Comment:
missing `--share=ipc` (need to x11 performance)
why does it need `home` access?

Reply:
for config and saving files

Reply:
if it uses the standard XDG directory (and it should) you don't need any permission.

Reply:
> but it needs the user to be able to save and open files from a gtk dialog. 

What does it use for filechooser? If it supports the portal then you need nothing.

Reply:
It uses filechooser for saving and opening of user projects files. It expects (for now at least) to save and load user settings in ~/.config/vht and if it can't it will fail at start. It doesn't really need access to home itself but it just looks better for the user integration-wise imo. It also gets default instrument/controller definition files from the data directory, which (more advanced) users can later edit to suit their needs. Not saying it has to stay that way, just how to make it run in current state.

Reply:
> It uses filechooser for saving and opening of user projects files. 

GtkFilechooserNative ? 

> ~/.config/vht

How is the directory calculated? Relative to `$HOME` or $XDG_CONFIG_HOME`?

Reply:
.config dir was wrong - fixed it - it now takes it from GLib.get_user_config_dir()

It uses Gtk.FileChooserDialog and seems to save and open files fine, but after
I close the program and reopen, the files are "gone". Maybe it's just me doing it wrong.

Reply:
> Gtk.FileChooserDialog

If that is gtk3, it does not support portals if you switch to GtkFilechooserNative it will work without any `--filesystem=` access.

Reply:
but then I have the same issue using qpwgraph so maybe it is just me...

Reply:
Native dialog did the trick. Thank you!

Reply:
Thank you for getting me to remove those unneeded perms.
I got nothing to add. Do I click something now?